packages/disploy/src/structs/managers/ChannelManager.ts

```diff
@@ -0,0 +1,37 @@
+import { ChannelType, RESTGetAPIChannelResult, Routes } from 'discord-api-types/v10';
+import type { App } from '../../client';
+import type { DiscordChannel } from '../../types';
+import { Base } from '../Base';
+import { GuildTextChannel } from '../GuildTextChannel';
+import { GuildVoiceChannel } from '../GuildVoiceChannel';
+
+export class ChannelManager extends Base {
+	private guildId?: string;
+
+	/**
+	 * A manager for fetching channels.
+	 * @param app
+	 * @param guildId The ID of the guild to lock the manager to.
+	 */
+	public constructor(app: App, guildId?: string) {
+		super(app);
+		this.guildId = guildId;
+	}
+
+	public async fetch(id: string): Promise<DiscordChannel> {
+		const raw = await this.app.rest.get<RESTGetAPIChannelResult>(Routes.channel(id));
+
+		if (this.guildId !== undefined && 'guild_id' in raw && raw.guild_id !== this.guildId) {
+			throw new Error(`Channel is not in the guild (${this.guildId})`);
+		}
+
+		switch (raw.type) {
+			case ChannelType.GuildText:
+				return new GuildTextChannel(this.app, raw);
+			case ChannelType.GuildVoice:
+				return new GuildVoiceChannel(this.app, raw);
+			default:
+				throw new Error(`Unknown channel type: ${raw.type}`);
```

Comment:
Should we instead fail silently here? Discord can add new channel types in the future which would make this app crash.

Reply:
Yeah that sounds like a better idea.

Reply:
resolved in https://github.com/Disploy/disploy/pull/35/commits/e4836c89c0d340b506f1c190c5a299a8401066db

Reply:
decided to just return an object with the type field so if the user was switching through channel types, it just falls through.